ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -318,21 +318,21 @@ private void read(ChannelHandlerContext ctx, DataStreamRequestByteBuf request) {
       }
     } else {
       info = streams.get(key);
-      localWrite = info.getStream().thenApply(stream -> writeTo(buf, stream));
+      localWrite = info.getPrevious().get()
```

Comment:
It seems that only the local write will wait for previous but the remote writes won't.  Then, the remote writes may go out of order.

Reply:
Let me test it a little bit more.

Reply:
Good catch.  The write should also wait for the previous.

Reply:
@szetszwo remote writes will not go out of order without wait for previous. Because request was send by OrderedStreamAsync, and out.writeAsync was called by the same thread.

Reply:
> ...  out.writeAsync was called by the same thread.

We should not assume "the same thread" in async programming.   Anyway, let's merge this first.  We may work on it later.

Reply:
Agree, let me change it.